packages/jest-mock/src/index.ts

```diff
@@ -1277,6 +1311,116 @@ export class ModuleMocker {
     return mock;
   }
 
+  mockProperty<
+    T extends object,
+    P extends PropertyLikeKeys<T>,
+    V extends Required<T>[P],
+  >(object: T, propertyKey: P, value: V): MockedProperty {
+    if (!object) {
+      throw new Error(
+        `mockProperty could not find an object on which to replace ${String(
+          propertyKey,
+        )}`,
+      );
+    }
+
+    if (!propertyKey) {
+      throw new Error('No property name supplied');
+    }
+
+    if (typeof object !== 'object') {
+      throw new Error(
+        `Cannot mock property on a primitive value; ${this._typeOf(
+          object,
+        )} given`,
+      );
+    }
+
+    let descriptor = Object.getOwnPropertyDescriptor(object, propertyKey);
+    let proto = Object.getPrototypeOf(object);
+    while (!descriptor && proto !== null) {
+      descriptor = Object.getOwnPropertyDescriptor(proto, propertyKey);
+      proto = Object.getPrototypeOf(proto);
+    }
+    if (!descriptor) {
+      throw new Error(`${String(propertyKey)} property does not exist`);
+    }
+    if (!descriptor.configurable) {
+      throw new Error(`${String(propertyKey)} is not declared configurable`);
+    }
+
+    if (this._isAlreadyMocked(object, propertyKey)) {
+      throw new Error(
+        `Cannot mock the ${String(
+          propertyKey,
+        )} property because it is already mocked`,
+      );
+    }
+
+    if (descriptor.get !== undefined) {
+      throw new Error(
+        `Cannot mock the ${String(
+          propertyKey,
+        )} property because it has a getter`,
+      );
+    }
+
+    if (descriptor.set !== undefined) {
+      throw new Error(
+        `Cannot mock the ${String(
+          propertyKey,
+        )} property because it has a setter`,
+      );
+    }
+
+    if (typeof descriptor.value !== typeof value) {
```

Comment:
Why do the types need to match? It would be unfortunate if I can't mock a `null` value with a `string` instead.

Reply:
I'd say so. But maybe I'm misunderstanding what this does. In the end, I'd want to be able to replace a property with a different type. This seems a pretty standard use case (e.g. replacing a some value with `null`).

Reply:
I will remove this check.